26_Autoexposure/app_resources/luma_meter.comp.hlsl

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2024-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#include "nbl/builtin/hlsl/luma_meter/luma_meter.hlsl"
+#include "nbl/builtin/hlsl/bda/bda_accessor.hlsl"
+#include "app_resources/common.hlsl"
+
+[[vk::combinedImageSampler]] [[vk::binding(0, 0)]] Texture2D texture;
+[[vk::combinedImageSampler]] [[vk::binding(0, 0)]] SamplerState samplerState;
+
+[[vk::push_constant]] AutoexposurePushData pushData;
+
+using Ptr = nbl::hlsl::bda::__ptr < uint32_t >;
+using PtrAccessor = nbl::hlsl::BdaAccessor < uint32_t >;
+
+groupshared float32_t sdata[WorkgroupSize];
+
+struct SharedAccessor
+{
+    uint32_t get(const uint32_t index)
+    {
+        return sdata[index];
+    }
+
+    void set(const uint32_t index, const uint32_t value)
+    {
+        sdata[index] = value;
+    }
+
+    void workgroupExecutionAndMemoryBarrier()
+    {
+        nbl::hlsl::glsl::barrier();
+    }
+};
+
+struct TexAccessor
+{
+    float32_t3 get(float32_t2 uv) {
+        return texture.Sample(samplerState, uv).rgb;
+    }
```

Comment:
because the TexAcessor is supposed to "know" the colorspace of the input image, I'd rather have it return `float32_t` luma with the CIE XYZ luminance component extraction finished here already